traitsui/qt4/range_editor.py

```diff
@@ -80,7 +80,7 @@ class SimpleSliderEditor(BaseRangeEditor):
     #: High value for the slider range
     high = Any()
 
-    #: Formatting string used to format value and labels
+    #: Deprecated: This trait is no longer used.
```

Comment:
I was thinking this should become a property that mirrors the editors `factory_str` but I don't think that is needed / allowed? (ie an instance of an editor can't set its factory's trait). We ultimately had previously just set `self.format = factory.format` in the `init` method anyways.

I am worried there may be a change in behavior if a user was to try to set `format` on an instance of a `RangeEditor`.  e.g `my_editor = RangeEditor()`, `my_editor.format = "something"`.   Previously, because we had  been using `self.format` throughout instead of `string_value`, so this would actually be used.  Now, AFAICT, the best we can do is pass it as a kwarg to `string_value` if a user tries to set it on an instance, but that will no longer take priority if the factory has its trait set...

I don't think we want to change any behavior here (even if the current behavior is strange).  I am unsure how to best move forward.

@rahulporuri any thoughts on this?  I hope my explanation made some sense, if not we can discuss tomorrow at standup

Reply:
So initially I expected this Test to pass on master, but fail on this branch:
```
def test_format_behavior_change(self):
        # format trait has been deprecated in favor of format_str. However,
        # behavior should be unchanged.
        model = RangeModel()
        with self.assertWarns(DeprecationWarning):
            my_editor = RangeEditor(format="%s ...")
            my_editor.format = "%s"
            view = View(
                Item(
                    "float_value",
                    editor=my_editor
                )
            )
        tester = UITester()
        with tester.create_ui(model, dict(view=view)) as ui:
            float_value_field = tester.find_by_name(ui, "float_value")
            float_value_text = float_value_field.locate(Textbox())
            self.assertEqual(
                float_value_text.inspect(DisplayedText()), "0.1"
            )
 ```
 
 but it passes on both.  I then realized `my_editor` is always just an instance of the `traitsui.editors.RangeEditor` `EditorFactory`.  The actual `Editor` instances are only created when e.g. the `_get_simple_editor_class` method is called.  If I access the `Editor` directly, e.g. doing something like this:

 ```
def test_format_behavior_change(self):
        # format trait has been deprecated in favor of format_str. However,
        # behavior should be unchanged.
        model = RangeModel()
        view = View(
            Item(
                "float_value",
                editor=RangeEditor(format="%s ...")
            )
        )
        tester = UITester()
        with tester.create_ui(model, dict(view=view)) as ui:
            float_value_field = tester.find_by_name(ui, "float_value")
            float_value_field._target.format = "%s"
            float_value_text = float_value_field.locate(Textbox())
            self.assertEqual(
                float_value_text.inspect(DisplayedText()), "0.1"
            )
```
I expected it to pass on master and fail on this branch.  However, this fails on both... 
So, AFAICT, behavior is not changed by this PR.

Reply:
EDIT: see the recently added test

Reply:
Can you search for users of the `format` trait on this editor? I'm not even sure how one would go about specifically searching for such uses. Whether or not users exist would help us decide on whether or not we need to convert `format` into a property as well - and make getting/setting `format` raise warnings as well.

Reply:
I was unable to find any users trying to set format on the Editor instance directly.
I did see users setting it on the Factory object, ie `RangeEditor(format= ...)`, but that is covered by the use of property/deprecation in `traitsui/editors/range_editor.py`.

TBH I am not certain what the pattern would be to set it on the Editor instance.  I guess accessing it through `UIInfo`?

Reply:
From traitsUI docs: "Note that not every trait on every editor can react to changes: some values are only used at editor creation time; however all editors support dynamically changing the enabled, visible and invalid traits. This feature can sometimes allow developers to avoid having to create a custom Handler subclass."